fairroot/alignment/FairAlignmentHandler.cxx

```diff
@@ -100,15 +102,23 @@ void FairAlignmentHandler::AddAlignmentMatrices(const std::map<std::string, TGeo
 
 void FairAlignmentHandler::RecomputePhysicalAssmbBbox() const
 {
-    TObjArray* pPhysNodesArr = gGeoManager->GetListOfPhysicalNodes();
-
-    for (auto pPhysNode : TRangeDynCast<TGeoPhysicalNode>(pPhysNodesArr)) {
-        if (pPhysNode) {
-            auto pShapeAsb = dynamic_cast<TGeoShapeAssembly*>(pPhysNode->GetShape());
-            if (pShapeAsb) {
-                // Should reach here only if the original node was a TGeoShapeAssembly
-                pShapeAsb->ComputeBBox();
-            }
+    /// After alignment, forces the recomputation of the bounding box of all volumesAssemblies and the rebuilding of
+    /// Nodes requiring it.
+    /// => Not done after each change to geometry in ROOT for perf reason
+    /// => to be called once all changes done
+    /// => original snippet proposed in https://github.com/root-project/root/issues/12242 by A. Gheata
+    TObjArray* volumes = gGeoManager->GetListOfVolumes();
+    for (auto vol : TRangeDynCast<TGeoVolume>(volumes)) {
+        if (!vol) {
+            continue;
+        }
+        if (vol->IsAssembly()) {
+            vol->GetShape()->ComputeBBox();
+        }
+        auto finder = vol->GetVoxels();
+        if (finder && finder->NeedRebuild()) {
+            finder->Voxelize();
```

Comment:
I was curious what this `Voxelize()` function does and it looks to me as if [it computes the bounding box a second time](https://github.com/root-project/root/blob/454a6dff73228a35d97bbfe84a29a6a08d53de84/geom/geom/src/TGeoVoxelFinder.cxx#L2229) in that case? Is this intended?

Reply:
Indeed. I must confess I did not check carefully what was done under the hood in the ROOT calls as it fixed our new case.

From what we had in CBM, I would suspect that the two lines before this block in this PR are catching all Assemblies including those for which the `NeedRebuild`/`NeedRefresh` flag was not set, while the `Voxelize` is applied only to those where it was set but also having an affect on non-Assembly volumes.

But the best is probably to ask this from @agheata in the [ROOT issue 12242](https://github.com/root-project/root/issues/12242), as he is the author of both the `Voxelize` block and the patch applied here, plus the fact that the answer may be interesting also for the broader ROOT audience.

Reply:
Side note, there is also a `TGeoVolume::Voxelize()` which seems to abort if the volume does not have a `TGeoVoxelFinder` (the thing returned by `GetVoxels()`) ... To me, the ROOT APIs look very confusing here.